src/nvim/ops.c

```diff
@@ -2524,6 +2528,58 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = dict_alloc();
+
+  // the yanked text
+  list_T *list = list_alloc();
+  for (int i = 0; i < reg->y_size; i++) {
+    list_append_string(list, reg->y_array[i], -1);
+  }
+  dict_add_list(dict, "regcontents", list);
+
+  // the register type
+  buf[0] = NUL;
+  buf[1] = NUL;
+  switch (reg->y_type) {
+  case MLINE: buf[0] = 'V'; break;
```

Comment:
Indentation of switch is invalid: `case` should be indented: https://neovim.io/develop/style-guide.xml?showone=Loops_and_Switch_Statements#Loops_and_Switch_Statements. Old code was converted to its view by who knows what and you should not copy indentation from it.